### PR TITLE
Updates to DevOps APi and Verify ReadMe

### PR DESCRIPTION
**Pull request by chidozieononiwu**
- Add logic for retrieving package infro from DevOps feed
- Add yml pipeline for Verify Readme Run

### CODE REVIEW

File: eng/common/scripts/Invoke-DevOpsAPI.ps1 [hunk 1]
```diff
@@ -158,5 +158,64 @@ function Add-RetentionLease {
...
+    $IncludeAllVersions="false",
+    [Parameter(Mandatory = $true)]
+    $PackageName,
+    $Base64EncodedAuthToken
+  )
+
+  $packages = Get-PackagesInArtifactFeed -Organization $Organization -Project $Project -FeedId $FeedId `
+    -ApiVersion $ApiVersion -IncludeAllVersions $IncludeAllVersions -Base64EncodedAuthToken $Base64EncodedAuthToken
+
+  return $packages.Where({$_.name -eq $PackageName})
```
Comment (benbp) [inline eng/common/scripts/Invoke-DevOpsAPI.ps1:201]: I think you can avoid querying ALL the packages (and the resulting extra time it may take) by using the `packageNameQuery` URI parameter instead (from [docs](https://docs.microsoft.com/en-us/rest/api/azure/devops/artifacts/artifact-details/get-packages?view=azure-devops-rest-6.0)).

File: eng/common/scripts/Invoke-DevOpsAPI.ps1 [hunk 2]
```diff
@@ -158,5 +158,64 @@ function Add-RetentionLease {
...
+    [Parameter(Mandatory = $true)]
+    $FeedId,
+    $ApiVersion="6.1-preview.1",
+    $IncludeAllVersions="true",
+    [Parameter(Mandatory = $true)]
+    $PackageName,
+    $Base64EncodedAuthToken
+  )
 
+  $package = Get-PackageInArtifactFeed -Organization $Organization -Project $Project -FeedId $FeedId `
```
Comment (benbp) [inline eng/common/scripts/Invoke-DevOpsAPI.ps1:217]: Related to my above comment, I'm not sure you need all these functions and the resulting parameter duplication if all you need is the package version, which you can get by making one API call with a package name query parameter and parsing that result instead.

File: eng/common/scripts/Invoke-DevOpsAPI.ps1 (hunk 1 above)
Comment (chidozieononiwu) [reply eng/common/scripts/Invoke-DevOpsAPI.ps1:201]: I actually need only the first function so I have removed the rest.

File: eng/common/pipelines/templates/steps/verify-readme.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
+parameters:
+  ScanPath: $(Build.SourcesDirectory)
```
Comment (weshaggard) [inline eng/common/pipelines/templates/steps/verify-readme.yml:2]: This looks unrelated.

File: eng/common/scripts/Invoke-DevOpsAPI.ps1 [hunk 4]
```diff
@@ -158,5 +158,26 @@ function Add-RetentionLease {
...
+    $Organization="azure-sdk",
+    $Project="public",
+    [Parameter(Mandatory = $true)]
+    $FeedId,
+    $ApiVersion="6.1-preview.1",
+    $IncludeAllVersions="false",
+    $Base64EncodedAuthToken
+  )
+
+  $uri = "https://feeds.dev.azure.com/$Organization/$Project/_apis/packaging/Feeds/$FeedId/packages?api-version=$ApiVersion&includeAllVersions=$IncludeAllVersions"
```
Comment (weshaggard) [inline eng/common/scripts/Invoke-DevOpsAPI.ps1:174]: Did you see my other feedback in your .NET PR? If we can avoid it lets not try pulling all the package versions when we can just use the package version min and max to get the package we want.

File: eng/common/scripts/Verify-Readme.ps1 [hunk 5]
```diff
@@ -1,15 +1,23 @@
 # Wrapper Script for Readme Verification
 [CmdletBinding()]
 param (
-  [string]$DocWardenVersion = "0.7.1",
-
+  [string]$DocWardenVersion = "0.7.2",
```
Comment (weshaggard) [inline eng/common/scripts/Verify-Readme.ps1:4]: I assume we should either just default the version either here or in the yml but not both.

File: eng/common/scripts/Invoke-DevOpsAPI.ps1 (hunk 4 above)
Comment (sima-zhu) [reply eng/common/scripts/Invoke-DevOpsAPI.ps1:174]: Or could you default to latest one if no feed version passed in?

File: eng/common/scripts/Invoke-DevOpsAPI.ps1 [hunk 6]
```diff
@@ -158,5 +158,26 @@ function Add-RetentionLease {
...
           -ContentType "application/json"
+}
+
+function Get-PackagesInArtifactFeed {
+  param (
+    $Organization="azure-sdk",
+    $Project="public",
+    [Parameter(Mandatory = $true)]
+    $FeedId,
+    $ApiVersion="6.1-preview.1",
```
Comment (sima-zhu) [inline eng/common/scripts/Invoke-DevOpsAPI.ps1:169]: Why we use preview version by default?

File: eng/common/scripts/Invoke-DevOpsAPI.ps1 [hunk 7]
```diff
@@ -158,5 +158,26 @@ function Add-RetentionLease {
...
+
+  $uri = "https://feeds.dev.azure.com/$Organization/$Project/_apis/packaging/Feeds/$FeedId/packages?api-version=$ApiVersion&includeAllVersions=$IncludeAllVersions"
+
+  $packages = Invoke-RestMethod `
+    -Method GET `
+    -Uri $uri `
+    -Headers (Get-DevOpsApiHeaders -Base64EncodedToken $Base64EncodedAuthToken) `
+    -MaximumRetryCount 3
 
+  return $packages.value
```
Comment (sima-zhu) [inline eng/common/scripts/Invoke-DevOpsAPI.ps1:182]: I am curious what we returned here. Could you add some comments and example?

File: eng/common/scripts/Invoke-DevOpsAPI.ps1 (hunk 6 above)
Comment (chidozieononiwu) [reply eng/common/scripts/Invoke-DevOpsAPI.ps1:169]: No particular reason. It apears to be the latest release.

File: eng/common/scripts/Invoke-DevOpsAPI.ps1 (hunk 7 above)
Comment (chidozieononiwu) [reply eng/common/scripts/Invoke-DevOpsAPI.ps1:182]: The ap returns an object like @{count, value}

File: eng/common/scripts/Invoke-DevOpsAPI.ps1 (hunk 4 above)
Comment (chidozieononiwu) [reply eng/common/scripts/Invoke-DevOpsAPI.ps1:174]: The challenge is to find the latest alpha version in the feed. Since sometime other versions other than alpha is listed as the latest thats why I need to retrieve all the abvailable versions.

Comment (chidozieononiwu) [reply eng/common/scripts/Invoke-DevOpsAPI.ps1:174]: I can alternatively get all versions for the specific package, but I will have to call the api numerous times.

Comment (weshaggard) [reply eng/common/scripts/Invoke-DevOpsAPI.ps1:174]: @chidozieononiwu I'd like to better understand why my suggestion at https://github.com/Azure/azure-sdk-for-net/pull/25503#discussion_r765146174 doesn't work before we move forward with this new API.

Comment (chidozieononiwu) [reply eng/common/scripts/Invoke-DevOpsAPI.ps1:174]: I tried it and it is very unreliable. I installes some packages and fails to install other which are clearly present in the feed. I can not say why these failures occur.

Comment (weshaggard) [reply eng/common/scripts/Invoke-DevOpsAPI.ps1:174]: I just posted some updates to that code in your other PR. Please give that a try.

Also I would like us to avoid taking a dependency on the devops api directly as I'd prefer we use general nuget feed apis so we could move to another nuget feed, l like nuget if we needed to.
